crates/router/src/core/webhooks/outgoing_v2.rs

```diff
@@ -0,0 +1,817 @@
+                state,
+                business_profile,
+                &cloned_key_store,
+                event,
+                request_content,
+                delivery_attempt,
+                Some(content),
+            ))
+            .await;
+        }
+        .in_current_span(),
+    );
+
+    Ok(())
+}
+
+#[allow(clippy::too_many_arguments)]
+#[instrument(skip_all)]
+pub(crate) async fn trigger_webhook_and_raise_event(
+    state: SessionState,
+    business_profile: domain::Profile,
+    merchant_key_store: &domain::MerchantKeyStore,
+    event: domain::Event,
+    request_content: OutgoingWebhookRequestContent,
+    delivery_attempt: enums::WebhookDeliveryAttempt,
+    content: Option<api::OutgoingWebhookContent>,
+) {
+    logger::debug!(
+        event_id=%event.event_id,
+        idempotent_event_id=?event.idempotent_event_id,
+        initial_attempt_id=?event.initial_attempt_id,
+        "Attempting to send webhook"
+    );
+
+    let merchant_id = business_profile.merchant_id.clone();
+    let trigger_webhook_result = trigger_webhook_to_merchant(
+        state.clone(),
+        business_profile,
+        merchant_key_store,
+        event.clone(),
+        request_content,
+        delivery_attempt,
+    )
+    .await;
+
+    let _ = raise_webhooks_analytics_event(
+        state,
+        trigger_webhook_result,
+        content,
+        merchant_id,
+        event,
+        merchant_key_store,
+    )
+    .await;
+}
+
+async fn trigger_webhook_to_merchant(
+    state: SessionState,
+    business_profile: domain::Profile,
+    merchant_key_store: &domain::MerchantKeyStore,
+    event: domain::Event,
+    request_content: OutgoingWebhookRequestContent,
+    delivery_attempt: enums::WebhookDeliveryAttempt,
+) -> CustomResult<(), errors::WebhooksFlowError> {
+    let webhook_url = get_webhook_url_from_business_profile(&business_profile)?;
+
+    let event_id = event.event_id;
+
+    let headers = request_content
+        .headers
+        .into_iter()
+        .map(|(name, value)| (name, value.into_masked()))
+        .collect();
+    let request = services::RequestBuilder::new()
+        .method(services::Method::Post)
+        .url(&webhook_url)
+        .attach_default_headers()
+        .headers(headers)
+        .set_body(RequestContent::RawBytes(
+            request_content.body.expose().into_bytes(),
+        ))
+        .build();
+
+    let response = state
+        .api_client
+        .send_request(&state, request, Some(OUTGOING_WEBHOOK_TIMEOUT_SECS), false)
+        .await;
+
+    metrics::WEBHOOK_OUTGOING_COUNT.add(
+        &metrics::CONTEXT,
+        1,
+        &[metrics::KeyValue::new(
+            MERCHANT_ID,
+            business_profile.merchant_id.get_string_repr().to_owned(),
+        )],
+    );
+    logger::debug!(outgoing_webhook_response=?response);
+
+    match delivery_attempt {
+        enums::WebhookDeliveryAttempt::InitialAttempt => match response {
```

Comment:
What is difference between `InitialAttempt` and `ManualRetry`?

Instead matching first on `delivery_attempt`, first match on `response`. Since current approach has more duplication than the vice-versa